prow/plugins/approve/approvers2/owners.go

```diff
@@ -0,0 +1,1003 @@
+	}
+
+	return filesToApprovers
+}
+
+// GetLeafApprovers returns a map from files -> people that are approvers for them (only the leaf)
+func (o Owners) GetLeafApprovers() map[string]sets.String {
+	ownersToApprovers := map[string]sets.String{}
+
+	for _, fn := range o.filenames {
+		ownersToApprovers[fn] = o.repo.LeafApprovers(fn)
+	}
+
+	return ownersToApprovers
+}
+
+// GetAllPotentialApprovers returns the people from relevant owners files needed to get the PR approved
+// It returns a list of people based on leaf owners files after subdirectories are ignored.
+// Example: If the PR has the following file changes:
+// 		- a/b/c.go
+//		- a/b/c/d.go
+//		- p/q.go
+// this function will return the list of approvers in a/b/c/OWNERS and p/OWNERS
+func (o Owners) GetAllPotentialApprovers() []string {
+	approvers := sets.NewString()
+	owners := o.GetOwnersSet()
+	for _, fn := range o.filenames {
+		if owners.Has(o.repo.FindApproverOwnersForFile(fn)) {
+			approvers.Insert(o.repo.LeafApprovers(fn).List()...)
+		}
+	}
+	approversOnly := approvers.List()
+	sort.Strings(approversOnly)
+	if len(approversOnly) == 0 {
+		o.log.Debug("No potential approvers exist. Does the repo have OWNERS files?")
+	}
+	return approversOnly
+}
+
+// GetReverseMap returns a map from people -> files. files for which they are an approver
+func (o Owners) GetReverseMap(approvers map[string]sets.String) map[string]sets.String {
+	approverFiles := map[string]sets.String{}
+	for file, approvers := range approvers {
+		for approver := range approvers {
+			if _, ok := approverFiles[approver]; ok {
+				approverFiles[approver].Insert(file)
+			} else {
+				approverFiles[approver] = sets.NewString(file)
+			}
+		}
+	}
+	return approverFiles
+}
+
+func findMostCoveringApprover(allApprovers []string, reverseMap map[string]sets.String, unapproved sets.String) string {
+	maxCovered := 0
+	var bestPerson string
+	for _, approver := range allApprovers {
+		filesCanApprove := reverseMap[approver]
+		if filesCanApprove.Intersection(unapproved).Len() > maxCovered {
+			maxCovered = len(filesCanApprove)
+			bestPerson = approver
+		}
+	}
+	return bestPerson
+}
+
+// temporaryUnapprovedFiles returns the list of files that wouldn't be
+// approved by the given set of approvers.
+func (o Owners) temporaryUnapprovedFiles(approvals []Approval) sets.String {
+	ap := NewApprovers(o)
+	for _, aprvl := range approvals {
+		for _, info := range aprvl.Infos {
+			ap.AddApprover(aprvl.Login, info.Reference, info.Path)
+		}
+	}
+	return ap.UnapprovedFiles()
+}
+
+// KeepCoveringApprovers finds who we should keep as suggested approvers given a pre-selection
+// knownApprovers must be a subset of potentialApprovers.
+func (o Owners) KeepCoveringApprovers(reverseMap map[string]sets.String, knownApprovals []Approval, potentialApprovers []string) sets.String {
+	if len(potentialApprovers) == 0 {
+		o.log.Debug("No potential approvers exist to filter for relevance. Does this repo have OWNERS files?")
```

Comment:
nit: Not sure what "exist to filter for relevance" means.

Reply:
This is copied from the original implementation. This might be a good candidate for a follow-up PR that covers boarder log statements improvements.